WhoIsThatServer.Storage/Helpers/TargetElementHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using WhoIsThatServer.Storage.Context;
+using WhoIsThatServer.Storage.Models;
+
+namespace WhoIsThatServer.Storage.Helpers
+{
+    public class TargetElementHelper : ITargetElementHelper
+    {
+        private IDatabaseContextGeneration _databaseContextGeneration;
+
+        public TargetElementHelper(IDatabaseContextGeneration databaseContextGeneration = null)
+        {
+            //If context is null new context will be created
+            _databaseContextGeneration = databaseContextGeneration ?? new DatabaseContextGeneration();
```

Comment:
But DB context will not be initialised? Its constructor after all, and it created new helper with null as DB context

Reply:
When helper's methods are called, DatabaseContextGeneration's method 'BuildDatabaseContext' is called which returns instance of DB context. Thought it was a good way, not sure though